python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/shape_lib_gen.py

```diff
@@ -589,6 +602,9 @@ def aten〇max〇dim(self: List[int], dim: int, keepdim: bool = False) -> Tuple[
     reduced_shape = _reduce_along_dim(self, dim, keepdim)
     return reduced_shape, reduced_shape
 
+def aten〇amax(self: List[int], dim: List[int] = (), keepdim: bool = False) -> List[int]:
+    return _reduce_along_dims(self, dim, keepdim)
```

Comment:
I would've expected `upstream_shape_functions.sum_mean_dim` to work here. Is that not the case?

Reply:
Yes, it works - I thought of following the norm for other surrounding ops which base their shape inference on a locally created function.

I believe as a follow-up action item I can create a "properly" named API in upstream PyTorch - for instance using `sum_mean_dim` for `amax` seems unnatural at the first glance. Let me know if that action item sounds good to you. :D

I've currently incorporated `upstream_shape_functions.sum_mean_dim` in this updated revision - you may take a look.

Reply:
Yeah, the `sum_mean_dim` name is a bit strange, but if a shape function can be expressed in terms of an upstream function, that's always preferred. In this case, `amax`, `sum`, and `mean` are all reduction functions, so it's not too odd to see `sum_mean_dim` in the shape function of `amax`. 

> I believe as a follow-up action item I can create a "properly" named API in upstream PyTorch - for instance using sum_mean_dim for amax seems unnatural at the first glance. Let me know if that action item sounds good to you. :D

Sure!